lib/curses_renderer.rb

```diff
@@ -16,35 +17,22 @@ def initialize(board)
     @cpu_player = Cpu.new(board)
   end
 
-  def show_game_over_screen
-    Curses.addstr("\n\nGAME OVER!\n\n")
-    Curses.refresh
-    sleep(3)
-  end
-
   def play!
     initialize_curses
 
     loop do
+      break if @quitting
+
       render
 
       if @board.game_over?
-        show_game_over_screen
-        break
+        show_game_over_screen and break
       end
 
-      key = @screen.getch
-      if key == 'q' || key == 'Q'
-
-        break
-      elsif movement?(key)
-        update_selection(key)
-      elsif painting?(key)
-        @board.paint!(@selected[0], @selected[1])
-        @board.whose_turn_is_it?
-
-        @cpu_player.cpus_turn
-        @board.whose_turn_is_it?
+      if @board.cpu_turn?
+        @cpu_player.take_turn
+      else
+        take_player_turn
```

Comment:
To keep some sort of parallelism I made each of these a single line — one line for the CPU to take their turn, one line for the player to take their turn. So I extracted the player input handling to its own method. That necessitated a new way to break this loop, so there's a new state flag `@quitting`.

There are other ways to do that, but I've seen this before and it seems fine for now.